content/advanced/fine-grained-reactivity.mdx

```diff
@@ -145,6 +149,7 @@ function createSignal(initialValue) {
   }
   
   function setter(newValue) {
+    if (value === newValue) return;
```

Comment:
maybe worth adding a comment like "checking if the new value is actually different, to avoid notifying dependent effects and memos unnecessarily". But it's not strictly necessary.